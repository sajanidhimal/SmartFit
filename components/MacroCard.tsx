
import React from 'react';
import { View, Text } from 'react-native';
import { Svg, Circle } from 'react-native-svg';

interface MacroCardProps {
  title: string;
  current: number;
  goal: number;
  unit: string;
  bgColor: string;
  progressColor: string;
  textColor: string;
}

export default function MacroCard({
  title,
  current,
  goal,
  unit,
  bgColor,
  progressColor,
  textColor
}: MacroCardProps) {
  const size = 40;
  const strokeWidth = 4;
  const radius = (size - strokeWidth) / 2;
  const circum = radius * 2 * Math.PI;
  const svgProgress = 100 - (current / goal) * 100;

  return (
    <View className={`flex-1 p-4 rounded-lg items-center ${bgColor}`}>
      <View className="items-center justify-center mb-2">
        <Svg width={size} height={size} viewBox={`0 0 ${size} ${size}`} className="rotate-90">
          {/* Background Circle */}
          <Circle
            cx={size / 2}
            cy={size / 2}
            r={radius}
            stroke="#ffffff"
            strokeWidth={strokeWidth}
            fill="transparent"
          />
          
          {/* Progress Circle */}
          <Circle
            cx={size / 2}
            cy={size / 2}
            r={radius}
            stroke={progressColor}
            strokeWidth={strokeWidth}
            strokeDasharray={circum}
            strokeDashoffset={circum * svgProgress / 100}
            strokeLinecap="round"
            fill="transparent"
          />
        </Svg>
      </View>
      
      <Text className={`font-semibold text-base ${textColor}`}>{title}</Text>
      <Text className={`text-sm ${textColor}`}>
        {current}/{goal} {unit}
      </Text>
    </View>
  );
}