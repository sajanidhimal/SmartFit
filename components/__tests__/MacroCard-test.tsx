import React from 'react';
import renderer from 'react-test-renderer';
import MacroCard from '../MacroCard';

describe('MacroCard', () => {
  it('renders correctly with all props', () => {
    const tree = renderer.create(
      <MacroCard 
        title="Protein"
        current={65}
        goal={120}
        unit="g"
        bgColor="bg-blue-100"
        progressColor="#3b82f6"
        textColor="text-blue-800"
      />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });
  
  it('renders correctly when goal is reached', () => {
    const tree = renderer.create(
      <MacroCard 
        title="Carbs"
        current={200}
        goal={200}
        unit="g"
        bgColor="bg-green-100"
        progressColor="#22c55e"
        textColor="text-green-800"
      />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });
}); 