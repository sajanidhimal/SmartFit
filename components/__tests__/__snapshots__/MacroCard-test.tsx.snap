// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MacroCard renders correctly when goal is reached 1`] = `
<View
  className="flex-1 p-4 rounded-lg items-center bg-green-100"
>
  <View
    className="items-center justify-center mb-2"
  >
    <View
      className="rotate-90"
      height={40}
      viewBox="0 0 40 40"
      width={40}
    >
      <View
        cx={20}
        cy={20}
        fill="transparent"
        r={18}
        stroke="#ffffff"
        strokeWidth={4}
      />
      <View
        cx={20}
        cy={20}
        fill="transparent"
        r={18}
        stroke="#22c55e"
        strokeDasharray={113.09733552923255}
        strokeDashoffset={0}
        strokeLinecap="round"
        strokeWidth={4}
      />
    </View>
  </View>
  <Text
    className="font-semibold text-base text-green-800"
  >
    Carbs
  </Text>
  <Text
    className="text-sm text-green-800"
  >
    200
    /
    200
     
    g
  </Text>
</View>
`;

exports[`MacroCard renders correctly with all props 1`] = `
<View
  className="flex-1 p-4 rounded-lg items-center bg-blue-100"
>
  <View
    className="items-center justify-center mb-2"
  >
    <View
      className="rotate-90"
      height={40}
      viewBox="0 0 40 40"
      width={40}
    >
      <View
        cx={20}
        cy={20}
        fill="transparent"
        r={18}
        stroke="#ffffff"
        strokeWidth={4}
      />
      <View
        cx={20}
        cy={20}
        fill="transparent"
        r={18}
        stroke="#3b82f6"
        strokeDasharray={113.09733552923255}
        strokeDashoffset={51.83627878423159}
        strokeLinecap="round"
        strokeWidth={4}
      />
    </View>
  </View>
  <Text
    className="font-semibold text-base text-blue-800"
  >
    Protein
  </Text>
  <Text
    className="text-sm text-blue-800"
  >
    65
    /
    120
     
    g
  </Text>
</View>
`;
